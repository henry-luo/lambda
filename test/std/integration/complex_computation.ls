// Test: Complex Computation
// Category: integration
// Type: positive
// Expected: 55

// Test complex arithmetic computation
let a = 10
a
let b = 20
b
let c = 5
c
let result = (a * b) - (c * 9)
result
result
