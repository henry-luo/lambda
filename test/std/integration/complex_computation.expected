09:15:49 [NOTICE] Start transpiling test/std/integration/complex_computation.ls...
[script:list]
  [content:list[5]]
    no declare
    item:
    [let stam:null]
      declare:
      [assign expr:a:int]
        [primary expr:int,const:1]
          [const: int]
    item:
    [let stam:null]
      declare:
      [assign expr:b:int]
        [primary expr:int,const:1]
          [const: int]
    item:
    [let stam:null]
      declare:
      [assign expr:c:int]
        [primary expr:int,const:1]
          [const: int]
    item:
    [let stam:null]
      declare:
      [assign expr:result:int]
        [binary expr -.4:int]
          [primary expr:int,const:0]
            [binary expr *.5:int]
              [primary expr:int,const:1]
                [ident:a:int,const:1]
              [primary expr:int,const:1]
                [ident:b:int,const:1]
          [primary expr:int,const:0]
            [binary expr *.5:int]
              [primary expr:int,const:1]
                [ident:c:int,const:1]
              [primary expr:int,const:1]
                [const: int]
    item:
    [primary expr:int,const:0]
      [ident:result:int,const:0]
09:15:49 [NOTICE] compiling C code in 'test/std/integration/complex_computation.ls' to MIR
C2MIR init end           -- 69 usec
test/std/integration/complex_computation.ls:1:2: warning -- unknown pragma
    preprocessor tokens -- 2436, parse tokens -- 1464
  C2MIR preprocessor end    -- 494 usec
  C2MIR parser end          -- 619 usec
  C2MIR context checker end -- 698 usec
  C2MIR generator end       -- 758 usec
C2MIR compiler end                -- 789 usec
09:15:49 [NOTICE] Generating native code...
##### Script 'test/std/integration/complex_computation.ls' executed: #####
155
